lib/vehicle_factory.rb

```diff
@@ -0,0 +1,28 @@
+class VehicleFactory
+  attr_reader :created_vehicles 
+  
+  def initialize
```

Comment:
I understand why you create this `initialize` method because you created an instance variable to hold the newly created objects but it is not necessary. 
```ruby
factory = VehicleFactory.new 
```
Has no arguments. So see my comments below on how you can just use the `create_vehicles` method to create new `Vehicle` objects without having to shovel them into a collector.